packages/next/src/build/next-config-ts/utils.ts

```diff
@@ -0,0 +1,71 @@
+import type { Options as SWCOptions } from '@swc/core'
+import type { CompilerOptions } from 'typescript'
+
+import { resolve } from 'node:path'
+import { parseJsonFile } from '../load-jsconfig'
+
+export function resolveSWCOptions(cwd: string): SWCOptions {
+  const tsConfig = lazilyGetTSConfig(cwd)
+
+  return {
+    jsc: {
+      parser: {
+        syntax: 'typescript',
+      },
+      paths: tsConfig?.compilerOptions?.paths,
+      // SWC requires `baseUrl` to be passed when `paths` are used.
+      // Also, `baseUrl` must be absolute.
+      baseUrl: resolve(cwd, tsConfig?.compilerOptions?.baseUrl ?? ''),
+      experimental: {
+        keepImportAttributes: true,
+        emitAssertForImportAttributes: true,
+      },
+    },
+    env: {
+      targets: {
+        node: process.versions.node,
+      },
+    },
+  } satisfies SWCOptions
+}
+
+// Since we only need "paths" and "baseUrl" from tsconfig for now,
+// we lazily look for tsconfig.json at cwd. Does not cover edge cases
+// like "extends" or even the case where tsconfig.json does not exist.
+export function lazilyGetTSConfig(cwd: string): {
+  compilerOptions?: CompilerOptions
+} {
+  let tsConfig = {}
+  try {
+    tsConfig = parseJsonFile(resolve(cwd, 'tsconfig.json'))
+  } catch (error) {
+    // ignore if tsconfig.json does not exist
+    if ((error as NodeJS.ErrnoException).code !== 'ENOENT') {
+      throw error
+    }
+  }
+
+  return tsConfig
+}
+
+// A bare specifier is a specifier that is not a relative path or a URL.
+// e.g. "next", "react", etc.
+export function isBareSpecifier(specifier: string) {
```

Comment:
Still not clear about this, the below try catch could error on a lot of cases. From the usage side it looks like it can be `next/xxx/xx` not just `next` or `react`

can you give example of input and output in comments like:
```
"...." => true
"xxx" => false
```

Reply:
Updated comments!

Reply:
Sounds like the implementation is just `return !specifier.startsWith('.') && !URL.canParse(specifier)`?

Would be more direct if we can have the comments like
```
// @example
// - Relative specifiers like './foo.js' => false
// - Absolute specifiers like 'file:///path/to/module' => false
// - Bare specifiers like 'next' or 'next/dist/compiled/react' => true
```

Reply:
Sounds perfect!

Reply:
Updated!